File: lib_audio/src/main/java/com/imooc/lib_audio/app/AudioHelper.java
package com.imooc.lib_audio.app;

import android.app.Activity;
import android.content.Context;

import com.imooc.lib_audio.mediaplayer.core.AudioController;
import com.imooc.lib_audio.mediaplayer.core.MusicService;
import com.imooc.lib_api.model.AudioBean;
import com.imooc.lib_audio.mediaplayer.view.MusicPlayerActivity;

import java.util.ArrayList;

//唯一与外界通信的类
public final class AudioHelper {

	private static Context mContext;

	public static void init(Context context){
		mContext = context;
		MusicService.startMusicService();
	}

	public static Context getContext(){
		return mContext;
	}


	//TODO AudioBean改造
	public static void addAudio(Activity activity, AudioBean bean){
		AudioController.getInstance().addAudio(bean);
		if(activity != null){
			MusicPlayerActivity.start(activity);
		}
	}

	//不启动Activity
	public static void addAudio(AudioBean bean){
		addAudio(null, bean);
	}

	//加入List的音乐
	public static void addAudio(ArrayList<AudioBean> mLists){
		AudioController.getInstance().addAudio(mLists);
	}

	//获取当前播放的歌曲
	public static AudioBean getNowPlaying(){
		return AudioController.getInstance().getNowPlaying();
	}

	//获取当前播放队列的歌曲
	public static ArrayList<AudioBean> getCurrentQueue(){
		return AudioController.getInstance().getQueue();
	}

}

File: lib_audio/src/main/java/com/imooc/lib_audio/app/AudioHelper.kt
package com.imooc.lib_audio.app

import android.content.Context
import com.imooc.lib_audio.mediaplayer.core.AudioController
import com.imooc.lib_audio.mediaplayer.core.MusicService
import com.imooc.lib_audio.mediaplayer.db.GreenDaoHelper
import com.imooc.lib_audio.mediaplayer.model.AudioBean
import com.imooc.lib_audio.mediaplayer.view.MusicPlayerActivity
import com.imooc.lib_common_ui.base.ProxyActivity

public class AudioHelper {


    companion object{
        private lateinit var mContext : Context

        public fun init(context:Context){
            mContext = context
            GreenDaoHelper.initDatabase()
            MusicService.startMusicService()
        }

        public fun getContext():Context{
            return mContext
        }

        //TODO AudioBean改造
        public fun addAudio(activity : ProxyActivity?, bean : AudioBean){
            AudioController.addAudio(bean)
            if(activity != null){
                MusicPlayerActivity.start(activity)
            }
        }

        public fun addAudio(bean : AudioBean){

            addAudio(null, bean)
        }

        //播放全部音乐
        public fun addAudio(list: ArrayList<AudioBean>){
            AudioController.addAudio(list)
        }

    }

}